src/dmd/ctfeexpr.d

```diff
@@ -873,7 +873,9 @@ UnionExp pointerArithmetic(const ref Loc loc, TOK op, Type type, Expression eptr
         error(loc, "CTFE internal error: bad pointer operation");
         goto Lcant;
     }
-    if (indx < 0 || len < indx)
+    // We allow indx to be equal to the len: one position below the right bound.
+    // So we do for the left bound for backward iteration algorithms.
+    if (!(-1 <= indx && indx <= len))
```

Comment:
compare to the previous code you reverse the order of the operands, changed the constant and added a negation.
anyone familiar with the previous code will be mighty surprised.
maybe there is a clearer way of stating this condition?

Reply:
```suggestion
(indx < -1 || len < indx)
```